### PR TITLE
PIP-1627: Add mxresolv package

### PR DESCRIPTION
**Pull request by horkhe**

### CODE REVIEW

**Review by thrawn01 — APPROVED** [review]

**Review by censhin — APPROVED** [review]
LGTM

File: mxresolv/mxresolv.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+const (
+	cacheSize = 1000
+	cacheTTL  = 10 * clock.Minute
+)
+
+var (
+	errNullMXRecord   = errors.New("domain accepts no mail")
+	errNoValidMXHosts = errors.New("no valid MX hosts")
+
+	lookupResultCache *collections.LRUCache
```
Comment (vtopc) [inline mxresolv/mxresolv.go:26]: Global state - is bad. https://peter.bourgon.org/blog/2017/06/09/theory-of-modern-go.html

File: mxresolv/mxresolv.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+	lookupResultCache = collections.NewLRUCache(cacheSize)
+}
+
+// Lookup performs a DNS lookup of MX records for the specified hostname. It
+// returns a prioritised list of MX hostnames, where hostnames with the same
+// priority are shuffled. If the second returned value is true, then the host
+// does not have explicit MX records, and its A record is returned instead.
+//
+// It uses an LRU cache with a timeout to reduce the number of network requests.
+func Lookup(ctx context.Context, hostname string) ([]string, bool, error) {
```
Comment (vtopc) [inline mxresolv/mxresolv.go:42]: Why no named return?

File: mxresolv/mxresolv.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+	"context"
+	"math/rand"
+	"net"
+	"sort"
+	"strings"
+	"unicode"
+
+	"github.com/mailgun/holster/v4/clock"
+	"github.com/mailgun/holster/v4/collections"
+	"github.com/pkg/errors"
```
Comment (vtopc) [inline mxresolv/mxresolv.go:13]: You must use `github.com/mailgun/holster/v4/errors`.
`github.com/pkg/errors` is deprecated.

**Pull request merged**